### PR TITLE
Refactor product list url scheme

### LINKED ISSUE #163: Product List URL spec
**Store Home**
/Store

**Parts and Tools pages**
/Store/Parts
/Store/Tools

**Tool Category**
https://www.ifixit.com/Store/Tools/Cutting

**Parts for Device**
 * Device: https://www.ifixit.com/Device/MacBook_Pro_15%22_Unibody_Late_2011
 * New Product List: https://www.ifixit.com/Store/Parts/MacBook_Pro_15%22_Unibody_Late_2011
   * Note the device slug matches across /Device and /Store/Parts. This will come from Strapi's deviceTitle field.
 * Old Product List: https://www.ifixit.com/Store/Mac/MacBook-Pro-15%22-Unibody-Late-2011
   * Note that this uses /Store/Mac instead of /Store/Parts
 
**Product URL**
https://www.ifixit.com/Store/Mac/MacBook-Pro-15-Inch-Unibody-Early-2011-Mid-2012-Battery/IF161-107?o=3
Note that this uses /Store/Mac but it would work just as well with /Store/Parts. The important pieces is the final slug `IF161-107` which is what we actually use to decide which part to display. This won't conflict with a product list since it has 4 slugs instead of 3.

### PR DESCRIPTION
**Pull request by dhmacs**
closes #163 

To implement this new url scheme I need a new `type` field in the product list (enum with values: `parts`, `tools`, `marketing`), so that when I have to generate a breadcrumb or just link to a specific product list, I know exactly what path to generate for the product list. All newly created product lists default to `parts`. Tools categories need to have `tools` type.. All other product lists should have type `marketing` (e.g. A "black-friday-sales" product list with type `marketing` would be rendered under `Store/black-friday-sales`.

cc @sterlinghirsh 

## QA

1. Visit the Vercel PR preview (replace `vercel.app` with `cominor.com`)
2. Visit parts product list page
3. Verify that the new product list spec detailed in #163 is correctly implemented

### CODE REVIEW

File: frontend/pages/Store/Parts/index.tsx [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+   return (
+      <ProductListView
+         productList={productList}
+         searchContext={searchContext}
+      />
+   );
+};
+
+ProductListPage.getLayout = function getLayout(page, pageProps) {
+   pageProps.globalSettings.newsletterForm;
```
Comment (sterlinghirsh) [inline frontend/pages/Store/Parts/index.tsx:94]: Why is this here?

**Comment by sterlinghirsh** [comment]
CR :zap: with a question

File: frontend/pages/Store/Parts/index.tsx (hunk 1 above)
Comment (dhmacs) [reply frontend/pages/Store/Parts/index.tsx]: Thanks for noticing it.. This is a stray instruction and has been there for a while. I've just removed it now 👍

**Comment by sterlinghirsh** [comment]
CR :zap:

**Comment by jordycosta** [comment]
**Just had two questions:**

1. Should the new URL scheme show just for the `Choose a model of All Parts` section on Preview?
    <details>
    <summary> Toggle View </summary> <br>
    
    >capture_1
     </details>

---

From #163 

> New Product List: https://www.ifixit.com/Store/Parts/MacBook_Pro_15%22_Unibody_Late_2011
Note the device slug matches across /Device and /Store/Parts. This will come from Strapi's deviceTitle field.

>Old Product List: https://www.ifixit.com/Store/Mac/MacBook-Pro-15%22-Unibody-Late-2011
Note that this uses /Store/Mac instead of /Store/Parts

2. Should the new directory for this product shown above work on the Preview?
    <details>
    <summary> Toggle View </summary> <br>
    
    >product
     </details>

**Comment by dhmacs** [comment]
> 1. Should the new URL scheme show just for the `Choose a model of All Parts` section on Preview?

#163 says

> The important pieces is the final slug IF161-107 which is what we actually use to decide which part to display

@sterlinghirsh should product parts links be handled differently? 

<hr/>

> 2. Should the new directory for this product shown above work on the Preview?

I think the new scheme should have `/Store/Parts` as prefix, unless I've misunderstood. Is it right @sterlinghirsh ?

**Comment by sterlinghirsh** [comment]
I don't really understand the first question but I think it looks right. Parts links currently look like https://www.ifixit.com/Store/Tools/Plastic-Cards/IF145-101?o=1 . https://www.ifixit.com/Store/Product/IF145-101 redirects, though I think we should just use the url field in the product_en index. Let me know if I'm misunderstanding the question.

For part 2, using /Store/Parts/ instead of /Store/Mac is correct. I don't think it gets us anything to say Mac instead of Parts in this url, and just saying Parts keeps it simpler.

**Comment by dhmacs** [comment]
> I don't really understand the first question but I think it looks right. Parts links currently look like https://www.ifixit.com/Store/Tools/Plastic-Cards/IF145-101?o=1 . https://www.ifixit.com/Store/Product/IF145-101 redirects, though I think we should just use the url field in the product_en index. Let me know if I'm misunderstanding the question.

Gotcha! I'm going to refactor to use `url` field and update the PR 👍

**Comment by jordycosta** [comment]
> I don't really understand the first question but I think it looks right. Parts links currently look like https://www.ifixit.com/Store/Tools/Plastic-Cards/IF145-101?o=1 . https://www.ifixit.com/Store/Product/IF145-101 redirects, though I think we should just use the url field in the product_en index. Let me know if I'm misunderstanding the question.

It was just the product redirect (e.g. [https://www.ifixit.com/Store/Product/IF145-101](https://www.ifixit.com/Store/Product/IF145-101)) that was throwing me off because it wasn't the same as the Vercel preview links like those in the `Choose a model of All Parts` section

**Comment by jordycosta** [comment]
QA 🟢 

The product list spec detailed in #163 is now enabled ✔️ 

---

### Testing Procedure:
1. Visit the Vercel `✅ Preview` [here](https://react-commerce-git-refactor-product-list-url-scheme-ifixit.vercel.app/)
2. Replace `vercel.app` with `cominor.com` in the address bar
3. Observe that the new product list spec is active

---

### Screen Captures:
screenshot

**Comment by dhmacs** [comment]
I've refactored product urls and fixed a few things

cc @jordycosta @sterlinghirsh

**Comment by jordycosta** [comment]
The product list URL's listed in [the spec ](https://github.com/iFixit/react-commerce/issues/163) are still present and functional ✔️ 

---

**Should the URL's in the filterable products section show the old URL config?**

I noticed that it is now displaying as:
> https://cominor.com/Store/Mac/iMac-Intel-27-Inch-2012-2019-Adhesive-Strips/IF174-060?o=1

Instead of in the older preview:
> https://www.ifixit.com/Store/Product/xxxxxxxx

---

<details>
<summary> Toggle View </summary> <br>

record
</details>

**Comment by dhmacs** [comment]
@jordycosta 

> **Should the URL's in the filterable products section show the old URL config?**

Yes 👍

**Comment by jordycosta** [comment]
QA 🟢 

The product list URL's listed in [the spec](https://github.com/iFixit/react-commerce/issues/163) are still present and functional ✔️ 

Followed the testing procedure shown [here](https://github.com/iFixit/react-commerce/pull/227#issuecomment-1076833615)

**Comment by sctice-ifixit** [comment]
CR :flags:

**Comment by sterlinghirsh** [comment]
CR :zap:
